### DISCUSSION TITLE
[12 settembre 2024] Esercizio 3

### DISCUSSION THREAD

**Discussion by Nw4pS** [sapienzastudentsnetwork/psd2223#272] [Esami]
Si analizzi il circuito sequenziale in figura e si disegni la FSM corrispondente. 

<div align="center">

![image](https://github.com/user-attachments/assets/3f5ae8c2-8637-4572-a0f3-6cd86aa8317f)

</div>

<details closed>
    <summary> soluzione A-L </summary>

<div align="center">

<img src="https://github.com/user-attachments/assets/888d7cff-f9e9-43e6-9ca0-003c4e86d7db" width="70%" height="70%">

</div>
</details>

**Comment by FeddyLix17** [comment]
$$
    \Large Q_1' = Q_0 \quad Q_0' = x
$$

$$
    \Large z = \overline{Q_1}\overline{Q_0}x + Q_1\overline{Q_0}\overline{x} + Q_1Q_0x
$$

<div align="center">

<table>
    <tr>
        <th>PS</th>
        <th rowspan="22"> </th>
        <th>$\Large Q_1$</th>
        <th>$\Large Q_0$</th>
        <th rowspan="22"> </th>
        <th>$\Large x$</th>
        <th rowspan="22"> </th>
        <th>NS</th>
        <th rowspan="22"> </th>
        <th>$\Large Q_1'$</th>
        <th>$\Large Q_0'$</th>
        <th rowspan="22"> </th>
        <th>$\Large z$</th>
    </tr>
    <tr>
        <td align="center">R</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">R</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">R</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">S1</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">1</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">S1</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">S2</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">S1</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">1</td>
        <td align="center">S3</td>
        <td align="center">1</td>
        <td align="center">1</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">S2</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">R</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">1</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">S2</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">S1</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">S3</td>
        <td align="center">1</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">S2</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">S3</td>
        <td align="center">1</td>
        <td align="center">1</td>
        <td align="center">1</td>
        <td align="center">S3</td>
        <td align="center">1</td>
        <td align="center">1</td>
        <td align="center">1</td>
    </tr>
</table>

<img src="https://github.com/user-attachments/assets/f67b2027-5082-4b5f-9790-dd70e064d2c5" width="70%" height="70%">

</div>

> [!NOTE]
> La FSM non coincide con quella della soluzione ufficiale (trovo che sia comunque corretta, fatemi sapere se trovate qualche typo)
